Extension/src/abTesting.ts

```diff
@@ -61,6 +69,20 @@ export class ABTestSettings {
         return util.isNumber(this.settings.gotoDefIntelliSense) ? this.settings.gotoDefIntelliSense >= this.bucket.Value : true;
     }
 
+    public getMinimumVSCodeVersion(curExtension: PackageVersion): PackageVersion {
+        if (this.minimumVSCodeVersionDefault.size === 0) {
+            return minimumVSCodeVersionDefault;
+        }
+        for (const [extensionString, vscodeString] of this.minimumVSCodeVersionDefault) {
+            const extensionPackage: PackageVersion = new PackageVersion(extensionString);
+            const vscodePackage: PackageVersion = new PackageVersion(vscodeString);
+            if (!extensionPackage.isGreaterThan(curExtension, undefined)) {
```

Comment:
Something doesn't seem right here.  It looks like you return the vscode requirement of an older version of the extension (which we can assume has already been satisfied).  We would actually want the requirement of a newer version in the event that a new requirement was published.

I apologize now for recommending doing this as a Map.  I don't think the insiders update code even works in a way that is compatible with what we were trying to design.  I think it just tries to get the most recent insiders build.  In that case, minimumVSCodeVersion could just be a string with the current VS code version requirement.  That would simplify all of this logic down to just converting the string from cpptools.json into a PackageVersion object.